[REF] *: convert AceEditor to owl

aab-odoo opened the pull request:
This PR refactors the website AceEditor to owl. The wrapper
around the lib was defined in web_editor, and extended in website,
where it was used (single usecase). This PR thus introduces an
owl Component (ResourceEditor) to replace it, directly in website,
and specialized to the website usecase.

This thus allows to remove the legacy implementation, and a part
of the legacy compatibility layer (useWidget).

This also removes the last usecase of Widget and select2 library
in the backend bundle, which allows us to trim it down.

Part of task~3439226

robodoo commented:
[Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/139154).

kebeclibre commented on addons/website/static/src/components/ace_editor/ace_editor.xml:
```diff
@@ -0,0 +1,82 @@
+        </xpath>
+    </t>
+    
+    <t t-name="website.AceEditor">
```
(nitpicking)
AceEditor doesn't really use Ace anymore (CodeEditor does). proposals:
AssetsEditor or WebsiteResourcesEditor ?

FrancoisGe commented on addons/website/static/src/components/ace_editor/ace_editor.js:
```diff
@@ -1,141 +1,501 @@
+
+        this.editorRef = useRef("editor");
+
+        this.debug = odoo.debug;
```
env.debug

rdeodoo reviewed (approved):
As discussed, I just did a functional check, everything seems to be working as it should (multi ressource edit, multi website (COW), errors, prevention overlay etc).
I couldn't properly test the debug mode part (showing key in the select, showing extra scss files etc) as we have an issue ATM that we still need [to fix](https://www.odoo.com/web#id=3475574&menu_id=4720&cids=1&action=333&active_id=974&model=project.task&view_type=form) to correctly forward the debug param.

So it's LGTM functional wise, here is the few minors things I found (but which can be ignored AFAIC or changed later):
- After hitting "reset", the file is correctly reset but the "fa-save" icon is still shown, probably a wrong state being kept?
Note that re-opening the HTML Editor fix this
![image](https://github.com/odoo/odoo/assets/30048408/91a9c0ca-73a7-4d25-8174-e6e9d7c59767)

- The indentation and "categories" are a bit harder to notice / see now IMHO, but that can totally be improved later I guess.
See before:
![image](https://github.com/odoo/odoo/assets/30048408/0d1432da-f595-438e-b9bd-fc03fafe2c8a)
After:
![image](https://github.com/odoo/odoo/assets/30048408/75980bea-11ac-4c2e-bd22-1bd0d3aa08eb)

- The templates are now listed by their name and not their key anymore, which was nice to figure exactly the template you were editing / looking for.
It was also the only way right now to be able to see the view key in the "ace" editor as we have an issue with the debug mode which was the other way to see the key (in debug mode it should appear next to the name).
![image](https://github.com/odoo/odoo/assets/30048408/f334dbf7-7dfb-4453-bc32-bb57ef80d37b)
![image](https://github.com/odoo/odoo/assets/30048408/36c2bc7d-f668-4c3d-a52a-ec9ae1d9c850)

aab-odoo commented:
Thanks a lot for the testing @rdeodoo 

> * After hitting "reset", the file is correctly reset but the "fa-save" icon is still shown, probably a wrong state being kept?

Fixed

> * The indentation and "categories" are a bit harder to notice / see now IMHO, but that can totally be improved later I guess.

I didn't find a trivial way to indent sources inside groups. But the whole SelectMenu could benifit from a design pass. I think that when we introduced it, we basically tried to mock the design of select2. We could indeed ask the designers to look at it in the future.

> * The templates are now listed by their name and not their key anymore, which was nice to figure exactly the template you were editing / looking for.

Fixed

kebeclibre commented on addons/web/static/src/legacy/utils.js:
```diff
@@ -41,26 +31,6 @@ export function createWidgetParent(env) {
 
-export function useWidget(refName, widgetClass, params = []) {
-    const ref = useRef(refName);
-    const env = useEnv();
```
change in another commit, and the test removal too.

robodoo commented:
@aab-odoo @kebeclibre linked pull request(s) odoo/enterprise#49286 not ready. Linked PRs are not staged until all of them are ready.
